presto-main/src/main/java/com/facebook/presto/server/security/SpnegoFilter.java

```diff
@@ -165,7 +165,8 @@ public Principal getUserPrincipal()
                         return;
                     }
                 }
-                catch (GSSException e) {
+                catch (Exception e) {
+                    LOG.debug(e, "Authentication failed for token %s", parts[1]);
```

Comment:
If you want to log here, we need to throw away the original exception, which might be the right thing to do if this goes directly to the client:
```java
throw new ServletException("Unknown error occurred during authentication");